wsagent/che-core-api-debugger/src/main/java/org/eclipse/che/api/debugger/server/Debugger.java

```diff
@@ -0,0 +1,197 @@
+     *
+     * @throws DebuggerException
+     *      if any error occur
+     */
+    List<Breakpoint> getAllBreakpoints() throws DebuggerException;
+
+    /**
+     * Gets the current value of the given variable.
+     *
+     * @param variablePath
+     *      the path to the variable
+     * @return {@link Value}
+     * @throws DebuggerException
+     *      if any error occur
+     */
+    Value getValue(VariablePath variablePath) throws DebuggerException;
+
+    /**
+     * Sets the new value {@link Variable#getValue()} of the variable {@link Variable#getVariablePath()}.
+     *
+     * @param variable
+     *      the variable to update
+     * @throws DebuggerException
+     *      if any error occur
+     */
+    void setValue(Variable variable) throws DebuggerException;
+
+    /**
+     * Evaluates the given expression.
+     *
+     * @param expression
+     *      the expression to evaluate
+     * @return the result
+     * @throws DebuggerException
+     *      if any error occur
+     */
+    String evaluate(String expression) throws DebuggerException;
+
+    /**
+     * Performs step over action.
+     * When process stops then {@link org.eclipse.che.api.debugger.shared.dto.events.SuspendEvent} must be fired.
+     *
+     * @param action
+     *      contains specific parameters
+     * @throws DebuggerException
+     *      if any error occur
+     */
+    void stepOver(StepOverAction action) throws DebuggerException;
+
+    /**
+     * Performs step into action.
+     * When process stops then {@link org.eclipse.che.api.debugger.shared.dto.events.SuspendEvent} must be fired.
+     *
+     * @param action
+     *      contains specific parameters
+     * @throws DebuggerException
+     *      if any error occur
+     */
+    void stepInto(StepIntoAction action) throws DebuggerException;
+
+    /**
+     * Performs step out action.
+     * When process stops then {@link org.eclipse.che.api.debugger.shared.dto.events.SuspendEvent} must be fired.
+     *
+     * @param action
+     *      contains specific parameters
+     * @throws DebuggerException
+     *      if any error occur
+     */
+    void stepOut(StepOutAction action) throws DebuggerException;
+
+    /**
+     * Resume application is being debugged.
+     * When process stops then {@link org.eclipse.che.api.debugger.shared.dto.events.SuspendEvent} must be fired.
+     *
+     * @param action
+     *      contains specific parameters
+     * @throws DebuggerException
+     *      if any error occur
+     */
+    void resume(ResumeAction action) throws DebuggerException;
+
+    /**
+     * Dump values of local variables, fields and method arguments of the current frame.
+     *
+     * @return {@link StackFrameDump}
+     * @throws DebuggerException
+     *      if any error occur
+     */
+    StackFrameDump dumpStackFrame() throws DebuggerException;
+
+    /**
+     * Is used to send back any events to client.
+     */
+    interface DebuggerCallBack {
```

Comment:
Are you sure about the `DebuggerCallBack` interface name?  I think that `DebuggerCallback` is more common and probably better name